timelock-impl/src/main/java/com/palantir/atlasdb/timelock/LockResource.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock;
+
+import java.math.BigInteger;
+import java.util.Optional;
+import java.util.Set;
+
+import com.palantir.lock.HeldLocksGrant;
+import com.palantir.lock.HeldLocksToken;
+import com.palantir.lock.LockClient;
+import com.palantir.lock.LockRefreshToken;
+import com.palantir.lock.LockRequest;
+import com.palantir.lock.LockResponse;
+import com.palantir.lock.LockRpcClient;
+import com.palantir.lock.LockServerOptions;
+import com.palantir.lock.LockService;
+import com.palantir.lock.SimpleHeldLocksToken;
+
+public class LockResource implements LockRpcClient {
```

Comment:
why do you need `LockResource` is it not sufficient to make `LockServiceImpl` implement `LockRpcClient`?

Reply:
Discussed offline; the current approach is cleaner.